stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/models/ProgressRequestBody.kt

```diff
@@ -25,20 +29,43 @@ internal class ProgressRequestBody(
         val total = file.length()
         val buffer = ByteArray(DEFAULT_BUFFER_SIZE)
         var uploaded = 0L
+
         FileInputStream(file).use { fis ->
             var read: Int
-            val handler = Handler(Looper.getMainLooper())
             while (fis.read(buffer).also { read = it } != -1) {
-                handler.post {
-                    callback.onProgress((100 * uploaded / total))
-                }
-                uploaded += read.toLong()
                 sink.write(buffer, 0, read)
+                uploaded += read.toLong()
+                withCallback { onProgress((100 * uploaded / total)) }
+            }
+        }
+
+        writeCount++
+    }
+
+    /**
+     * Only do progress updates if we've already
+     * ignored enough writes of this body.
+     */
+    private inline fun withCallback(crossinline actions: ProgressCallback.() -> Unit) {
+        if (writeCount >= PROGRESS_UPDATES_TO_SKIP) {
+            handler.post {
+                callback.actions()
             }
         }
     }
 
     companion object {
         private const val DEFAULT_BUFFER_SIZE = 2048
+
+        /**
+         * A number of writes to ignore (not issue progress updates for).
+         *
+         * This accounts for the [HttpLoggingInterceptor] and [CurlInterceptor] configured in
+         * [BaseChatModule] that will trigger writing the body into their logs.
+         *
+         * We don't want to issue progress updates for these writes, only the real write that's
+         * going out to the network that happens after these.
+         */
+        private const val PROGRESS_UPDATES_TO_SKIP = 2
```

Comment:
I don't like this much, but there's no apparent other way to achieve this. Logging the body with these two interceptors triggers the same writes, and we somehow have to ignore those when reporting our progress.

This is also the best wisdom that the internet had on this https://stackoverflow.com/a/44125313

A real solution would be intercepting the call with an OkHttp network interceptor, but we can't pass in individual callbacks per different Retrofit upload calls with that approach (at least not easily).